include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -95,7 +96,8 @@ inline std::mutex& map_lock()
   return map_lock;
 }
 
-inline auto& get_map()
+// Must have default visibility, see: https://github.com/rapidsai/rmm/issues/826
+RMM_EXPORT inline auto& get_map()
```

Comment:
Don't you need this on all functions in this file?

Reply:
I tested it with XGBoost and it works fine.  I'm not entirely sure what's happening exactly during runtime linking.  I guess as long as this function is not inlined and the linker can find the right symbol it should be fine?